### DISCUSSION TITLE
Proposal: Add list of available filter types to spec

### DISCUSSION THREAD

**Discussion by SetTrend** [graphql/graphql-spec#1083] [Ideas]
Currently, the only data filters I know of is "filter by property (list)" and `first:`, `offset:` and `after:`.

It seems to me that the spec lacks a list of available filter functions. So, I'd like to suggest to add such reference page.

**Comment by martinbonnin** [comment]
Did you mean this for the [Relay specification](https://relay.dev/graphql/connections.htm#sec-Forward-pagination-arguments)? 

I don't think the GraphQL spec specifies anything about field arguments and this was probably on purpose to allow different implementations.

**Reply by SetTrend** [reply]
Thanks for replying.

> Did you mean this for the [Relay specification](https://relay.dev/graphql/connections.htm#sec-Forward-pagination-arguments)?

Presumably, yes. So, is pagination the only field where filter keywords have been specified by the spec?

**Reply by martinbonnin** [reply]
The GraphQL spec (this repo) has very little opinion on what argument may be and I don't think `filters` are any part of it. You'll have to look at other specs built on top of GraphQL like the relay one.

**Reply by SetTrend** [reply]
Ah, I understand now.

So, for (new) users to get things clear, wouldn't it be appropriate to be explicit about Relaying being an (optional) GraphQL extension? Particularly on this spec page: [https://graphql.org/learn/pagination/](https://graphql.org/learn/pagination/)?

**Reply by martinbonnin** [reply]
If you read to the end, there's a mention of [Relay specification](https://graphql.org/learn/pagination/#connection-specification). But I agree it could be more explicit what is spec vs what is docs. Maybe open an issue at https://github.com/graphql/graphql.github.io?

**Reply by SetTrend** [reply]
Done: https://github.com/graphql/graphql.github.io/issues/1626

Thank you for pointing me.
